students/thecthaeh/Lesson04/mailroom_part2.py

```diff
@@ -60,32 +58,47 @@ def handle_selection2():
     report = []
     
     #fill the report with donor information
-    for donor in donor_info:
-        report.append([donor[0], sum(donor[1]), len(donor[1]), sum(donor[1])/len(donor[1])])
+    for donor in donor_info_dict:
+        report.append([donor, sum(donor_info_dict[donor]), len(donor_info_dict[donor]), sum(donor_info_dict[donor])/len(donor_info_dict[donor])])
     
     #sort the report
-    sorted_report = sorted(report, key = itemgetter(1), reverse = True)
+    sorted_report = sorted(report, key=itemgetter(1), reverse=True)
     
     #print the sorted report
     for donor in sorted_report:
         print("{:30}  ${:>19.2f}  {:>15}  ${:>19.2f}".format(*donor[:]))
     
     prompt()
 
+#function for selection 3
+def option3():
+    letter_dict = {}
```

Comment:
why create this dictionary? you can do the same thing in just looping over the main data structure:
```
for name, donations in donor_info_dict.items():
    f.write("...".format(name, sum(donations))
```